### DISCUSSION TITLE
load only fp1 fp2 2023 laps and weather

### DISCUSSION THREAD

**Discussion by Qbsol** [theOehrly/Fast-F1#534] [Q&A]
Hi, need to some help how to load session.to get only: laps and weather data to all fp1 and fp2 session of 2023 ... now i push all weekend data to cache, don't want all but exactly what i need specially if its for all year ;/ probably must pick somehow weekend list of 2023 ... im newbie low code user but for hard users its probably basic case

**Comment by Casper-Guo** [comment]
Probably something like:

```
for i in range(1, 22):
   fp1 = ff1.get_session(2023, i, "FP1")
   fp2 = ff1.get_session(2023, i, "FP2")
   fp1.load(telemetry=False, messages=False)
   fp2.loda(telemetry=False, messages=False)
```

**Reply by JochemZ** [reply]
problem with this is that Azerbaidjan 2023 did not have a FP2 session, so this is giving an error

**Reply by theOehrly** [reply]
That is because that was a sprint weekend of course. So you'd have various options to deal with that. One way would be to just catch the exception and go on.
A more clean solution is to use ``get_event_schedule`` (https://docs.fastf1.dev/fastf1.html#fastf1.get_event_schedule) and iterate over the ``Event``s (rows) in the schedule. Each ``Event`` has an 'EventFormat' specified, so you can check that the weekend format is "conventional". Then you can use ``Event.get_session()`` to get the FP1 and FP2 for that event.

See:
https://docs.fastf1.dev/events.html#event-schedule-data
https://docs.fastf1.dev/events.html#fastf1.events.Event.get_session
